osie-runner/handlers.py

```diff
@@ -236,7 +236,7 @@ def cacher_to_metadata(j, tinkerbell):
         "hostname": instance["hostname"],
         "id": instance["id"],
         "network": {
-            "addresses": instance["ip_addresses"],
+            "addresses": instance.get("ip_addresses", []),
```

Comment:
I'm now very cautious about this with the data api is ultimately responsible for. I think we should walk the safe side and do
```python
addr = instance.get("ip_addresses")
if not addr:
    addr = []
```

Reply:
I believe the main problem, is hegel now has a struct for instance to be parsed into rather than the interface it got tossed into before. In order to be backwards compatible, we had to put `omitempty` on all the fields inside the instance struct so the result for an empty instance is `instance: {}` however, this now breaks in this case where the api may be pushing an empty array of ips, but because we have omitempty, it's not including that attribute in the output.

Also a one line example could be `"addresses": instance.get("ip_addresses") or [],` which is the same thing

Reply:
ah yeah I was wanting a one-liner and forgot about simple `or ...`